_published/0014-test-results-handling.md

```diff
@@ -0,0 +1,274 @@
+---
+layout: default
+# This is just to fool remark-stringify not to escape & symbols
+# See https://github.com/syntax-tree/mdast-util-to-markdown/issues/8
+shields_io_query_params: label=issue%20state&logo=github&style=flat-square
+
+# Update the following (it's YAML syntax)
+pr_id: 14 # Update this with PR number/ID. No leading zeros
+rfc_feature_name: rework-handling-of-test-results # Use kebab-case
+title: "RFC-0014: rework-handling-of-test-results" # Update this with PR number/ID and feature name. Use leading zeros
+rfc_author_username: Alexamakans
+rfc_author_name: Alexander Fougner # Or same as username, if you wish
+
+# Leave these. Collaborator changes this before merging
+impl_issue_id: 0
+impl_issue_repo: iver-wharf/wharf-api
+last_modified_date: YYYY-MM-DD
+---
+
+# {{page.title}}
+
+- RFC PR: [iver-wharf/rfcs#{{page.pr_id}}](https://github.com/iver-wharf/rfcs/pulls/{{page.pr_id}})
+- Feature name: `{{page.rfc_feature_name}}`
+- Author: {{page.rfc_author_name}} ([@{{page.rfc_author_username}}](https://github.com/{{page.rfc_author_username}}))
+- Implementation issue: [{{page.impl_issue_repo}}#{{page.impl_issue_id}}](https://github.com/{{page.impl_issue_repo}}/issues/{{page.impl_issue_id}})
+- Implementation status: ![GitHub issue state](https://img.shields.io/github/issues/detail/state/{{page.impl_issue_repo}}/{{page.impl_issue_id}}?{{page.shields_io_query_params}})
+
+## Summary
+
+The intent is to avoid filtering and parsing test result files (TRX)
+from the generated artifacts each time we want to display them or a
+summary.
+
+We also want to display more detailed test result summaries in wharf-web.
+For this we will need to also store the messages that the tests produce.
+At the very least for failed tests.
+
+## Motivation
+
+The Status field in wharf-web's project view (Failed/Completed)
+currently calls an endpoint that asks the server to fetch all TRX files
+for the build, and take the most recent one. Then it parses the file to
+see how many failed/completed tests there were.
+It does this for every build in the view, every time it is refreshed.
+This leads to unnecessary requests and computation. This, in turn, slows
+down the response times and increases the API load.
```

Comment:
(open question) Would you regard tackling the "n+1" issue related to this RFC or out-of-scope to this RFC?

When fetching the list of builds, we don't want to also fetch the test result summaries on a request per build, but instead get that info in the build list directly. Such as having the summed up summaries in the build object returned by the `GET /project/{projectid}/builds` endpoint

Reply:
Feels like it is within scope of the original idea of optimizing the process/flow. Although the Summary I've written is more focused on the filtering/parsing, that's more because I didn't consider it before.

Reply:
Showing a glimpse of the test summary in the builds list is one of the big requirements for this feature, so I think it would be good then to elaborate on that in the RFC